File: apps/server-go/internal/domain/watchlist.go
package domain

import "time"

type Watchlist struct {
	WatchlistID string    `json:"watchlist_id"`
	CustomerID  string    `json:"customer_id"`
	Name        string    `json:"name"`
	CreatedAt   time.Time `json:"created_at"`
}

type WatchlistItem struct {
	WatchlistItemID string    `json:"watchlist_item_id"`
	WatchlistID     string    `json:"watchlist_id"`
	InstrumentID    string    `json:"instrument_id"`
	AddedAt         time.Time `json:"added_at"`
}

File: apps/mobile/src/types/watchlist.ts
export interface Watchlist {
  watchlist_id: string;
  customer_id: string;
  name: string;
  created_at: string;
}

export interface WatchlistItem {
  watchlist_item_id: string;
  watchlist_id: string;
  instrument_id: string;
  added_at: string;
}

export interface WatchlistCreate {
  customer_id: string;
  name: string;
}

export interface WatchlistItemCreate {
  watchlist_id: string;
  instrument_id: string;
}
